src/wp-includes/l10n.php

```diff
@@ -755,6 +756,9 @@ function load_textdomain( $domain, $mofile ) {
 
 	$l10n[ $domain ] = &$mo;
 
+	/** @var WP_Textdomain_Registry $wp_textdomain_registry */
+	$wp_textdomain_registry->set( $domain, dirname( $mofile ) );
```

Comment:
This needs some review because the [patch](https://core.trac.wordpress.org/attachment/ticket/39210/39210.2.diff) has a few more (unclear) changes here, like removing `$l10n[$domain] = &$mo;`.

Reply:
Unfortunately I didn't really explain the change in https://core.trac.wordpress.org/ticket/39210#comment:11 😕 

I don't see the benefit of removing `$l10n[$domain] = &$mo;` at the moment, especially because we then have to maintain BC.